### PR TITLE
Istio phase 1

### PR DESCRIPTION
**Pull request by stepanstipl**
This PR is overarching for Istio rollout. The actual rollout is split into several phases (to reduce complexity and provide no-downtime deployment).

**This should not be merged before the certificate validation issue is resolved - [GPII-3781](https://issues.gpii.net/browse/GPII-3781)**

The whole Istio rollout consists of following steps and corresponding PRs (in order):
1. **Enable Istio sidecar for Preferences and Flowmanager services** (this PR)
   This step deploys Istio sidecar alongside services, without actually enabling TLS traffic nor moving away from current nginx-ingress.
2. **Enable TLS traffic between services** (phase 2 PR - #323 )
  This step enables TLS traffic (and mTLS) between services, without enforcing it.
3. **Change DNS to IngressGateway** (phase 3 PR - #324 )
  This step points external DNS records to IngressGateway.
4. **Enforece mTLS** (phase 4 PR - #325 )
  This step enables mTLS enforcement between services, once access via current nginx-ingress is not needed anymore.
5. **Cleanup** (cleanup PR TBD)
  This phase is to clean up and will include:
    - decommissioning current nginx-controller
    - disabling GKE ingress load-balancing add-on (currently enabled but not used)
    - cleanup of global IP address and wildcard DNS records

**This PR**
This PR deploys Istio sidecar alongside services, without actually enabling TLS traffic nor moving away from current nginx-ingress. It also changes health checks to exec, as Kubernetes http probes can't cope with mutual TLS authentication.

Included changes:
- Enable Istio sidecar injection on GPII namespace
- Add Istio namespace migration code
- Change Preferences and Flowmanager Health and Readiness checks to exec
- Disable Istio injection on CouchDB and nginx-ingress pods
- Clean up cluster migration code after GPII-3568
- Disable Istio mtls for connections to CouchDB
- Charts yaml formatting improvements for consistency
- Expose Flowmanager and Preferences via Istio ingressGateway
- Disable mTLS initially, till both services are on Istio
- Relax mTLS auth requirements initially for Preferences and Flowmanager
- Bump Flowmanager and Preferences Charts versions

After deployment, verify that:
1. both Preferences and Flowmanager still work
2. both services are running with Istio sidecar container
3. ~both services are~ Flowmanager is reachable over Istio's IngresGateway IP

### CODE REVIEW

File: shared/charts/gpii-flowmanager/templates/gateway.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    istio: ingressgateway
+  servers:
+  - port:
+      number: 443
+      name: https-{{ template "flowmanager.name" . }}
+      protocol: HTTPS
+    tls:
+      mode: SIMPLE
+      serverCertificate: /etc/istio/ingressgateway-certs/flowmanager-tls.crt
+      privateKey: /etc/istio/ingressgateway-certs/flowmanager-tls.key
```
Comment (mrtyler) [inline shared/charts/gpii-flowmanager/templates/gateway.yaml]: Where do these files come from? Are they managed by the injected istio sidecar container?

Comment (stepanstipl) [reply shared/charts/gpii-flowmanager/templates/gateway.yaml]: These certs come from **cert-manager** originally, and they're merged into `istio-ingressgateway-certs` secret in `istio-system` namespace by **certmerge-operator** (that was deployed previously). This secret is then mounted in the `istio-ingressgateway` pods. (See certmerge-operator chart and https://github.com/gpii-ops/gpii-infra/blob/master/gcp/modules/certmerge-operator-crd/values.yaml for more details)

**Comment by stepanstipl** [comment]
cool, thanks for the review

> I like the rollout plan. I'd like to be involved with the various phases of rollout, as Istio is about to become mission critical and no one but you has worked with it so far :).

Thanks, and ok - this should end up in no-downtime deployment, but we can do sessions for the push to prod & verification phases, that's a good idea. At some point I also wanted to show you `istioctl` command, which we din't get to in the inro, but this might be a good opportunity 👍

**Comment by amatas** [comment]
LGTM, I'd like to be present in that group deployment.

**Pull request merged**